### PR TITLE
Add basic test for --watch mode

### PR DESCRIPTION
**Pull request by RyanZim**
Can @ai or @michael-ciniawsky peer review this? Due to chokidar's nature, writing this test was a bit difficult, I want someone to make sure I didn't make a big mistake.

### CODE REVIEW

File: test/watch.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  .then(dir => {
+    // Init watcher:
+    var watcher = chokidar.watch('.', {
+      cwd: dir,
+      ignoreInitial: true,
+      awaitWriteFinish: true
+    })
+
+    // On the first output:
+    watcher.on('add', p => {
```
Comment (michael-ciniawsky) [inline test/watch.js]: Unrelated to `postcss-cli`

File: test/watch.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      if (p === 'out.css') {
+        isEqual(p, 'test/fixtures/a-red.css')
+        .then(() => read('test/fixtures/a-blue.css'))
+        .then(css => fs.writeFile(path.join(dir, 'a-red.css'), css))
+        .catch(done)
+      }
+    })
+
+    // When the change is picked up:
+    watcher.on('change', p => {
```
Comment (michael-ciniawsky) [inline test/watch.js]: Unrelated to `postcss-cli`

File: test/watch.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      if (p === 'out.css') {
+        isEqual(p, 'test/fixtures/a-blue.css')
+        .then(() => done())
+        .catch(done)
+      }
+    })
+
+    // Start postcss-cli:
+    watcher.on('ready', () => {
```
Comment (michael-ciniawsky) [inline test/watch.js]: 👍

**Comment by RyanZim** [comment]
> The first 2 test are not really related to postcss-cli

@michael-ciniawsky Not sure what you mean.

To clarify, the code is not executed in order, here is a breakdown of what it does:

1. Create a test environment.
1. When chokidar is ready, start postcss-cli.
1. When the add event is fired for the output file, assert that we have the correct output. (Since postcss-cli doesn't exit in watch mode, the add event is the only way we can tell when processing is completed.)
1. Change the input file contents.
1. When the change event is fired for the output file, check that we have the correct output for the new input.
1. Kill the postcss child process & exit.

I realize the test is rather complex, but I wanted a robust test that wouldn't give false results (the old `make` tests for watch mode relied on `sleep`ing long enough to allow processing to finish, and didn't always work)

> refactor with an postcss-cli/postcss-import/result.messages.dependency example later

I do plan on adding a test for dependency messages later; we still need a test to make sure watch is working in the first place!

**Comment by michael-ciniawsky** [comment]
So the 'change', 'add' gets executed after `postcss-cli` kicked off and especially the 'change' goes through `postcss-cli`?

**Comment by RyanZim** [comment]
@michael-ciniawsky Yeah, there isn't any other way to know when postcss-cli is done processing the input and writing the output other than watching the output file.

I just wanted a bit of review since there are so many code paths in one test.

I'd be ready to merge this.

**Comment by michael-ciniawsky** [comment]
@RyanZim  We can do it later if there is easier solution, I was also mistaken here. LTGM for now :shipit: 

I plan to release `postcss-load-config` with `config.file` support and better error handling today. Can I ping you on the error handling issue later for review ?

**Comment by RyanZim** [comment]
@michael-ciniawsky Yep, no problem.

**Pull request merged**